tornado_opentracing/_tracing.py

```diff
@@ -0,0 +1,176 @@
+# Copyright The OpenTracing Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import wrapt
+
+import opentracing
+from opentracing.ext import tags
+
+from .context_manager import trace_context
+from ._constants import SCOPE_ATTR
+
+
+class BaseTornadoTracing(object):
+    """
+    @param tracer the OpenTracing tracer to be used
+    to trace requests using this TornadoTracing
+    """
+    def __init__(self, tracer=None, start_span_cb=None):
+        if start_span_cb is not None and not callable(start_span_cb):
+            raise ValueError('start_span_cb is not callable')
+
+        self._tracer_obj = tracer
+        self._start_span_cb = start_span_cb
+        self._trace_all = False
+        self._trace_client = False
+
+    @property
+    def _tracer(self):
+        """
+        DEPRECATED
+        """
+        return self.tracer
+
+    @property
+    def tracer(self):
+        if self._tracer_obj is None:
+            return opentracing.tracer
```

Comment:
This is deprecated by `global_tracer()` but would require a fallback for 2.0 users anyway.  Since you're pinning to <2.4 there should be no harm, but may require changing in the future.